pex/pip.py

```diff
@@ -588,18 +611,179 @@ def create(
                             import runpy
+                            import sys
 
+                            skip_markers = os.environ.pop({skip_markers_env_var_name!r}, None)
                             patched_markers_file = os.environ.pop(
                                 {patched_markers_env_var_name!r}, None
                             )
-                            if patched_markers_file:
-                                def patch_markers():
+                            if skip_markers is not None and patched_markers_file is not None:
+                                raise AssertionError(
+                                    "Pex should never both set both {skip_markers_env_var_name} "
+                                    "and {patched_markers_env_var_name} environment variables."
+                                )
+
+                            if skip_markers:
+                                python_full_versions = []
+                                python_versions = []
+                                python_majors = []
+
+                                python_versions_file = os.environ.pop(
+                                    {python_versions_env_var_name!r}, None
+                                )
+                                if python_versions_file:
+                                    import json
+
+                                    with open(python_versions_file) as fp:
+                                        python_full_versions = json.load(fp)
+                                    python_versions = sorted(
+                                        set(
+                                            (version[0], version[1])
+                                            for version in  python_full_versions
+                                        )
+                                    )
+                                    python_majors = sorted(
+                                        set(version[0] for version in  python_full_versions)
+                                    )
+
+                                # 1.) Universal dependency environment marker applicability.
+                                #
+                                # Allows all dependencies in metadata to be followed regardless
+                                # of whether they apply to this system. For example, if this is 
+                                # Python 3.10 but a marker says a dependency is only for
+                                # 'python_version < "3.6"' we still want to lock that dependency
+                                # subgraph too.
+                                def patch_marker_evaluate():
+                                    from pip._vendor.packaging import markers
+
+                                    original_get_env = markers._get_env
+                                    original_eval_op = markers._eval_op
+                                    
+                                    skip = object()
```

Comment:
Would having empty `python_versions` or `python_full_versions` every result in "accidentally" skipping this behavior? Or should the empty set have the same behavior as skipping anyway?

Reply:
Yeah, same as skip - this is not accidental. An unconstrained (no `--interpreter-constraint`) universal resolve should skip (ignore) `python_version` and `python_full_version` environment markers, and resolve every single wheel available. There is a test for this in this PR with 5 wheels downloaded for the `--interpreter-constraint` supplied case (`python_full_versions` not empty) and 31 wheels for the `--interpreter-constraint` not present case (`python_full_versions` is empty).